site/app/controllers/PollController.php

```diff
@@ -0,0 +1,236 @@
+
+        return MultiResponse::RedirectOnlyResponse(
+            new RedirectResponse($this->core->buildCourseUrl(['polls']))
+        );
+    }
+
+    /**
+    * @Route("/{_semester}/{_course}/polls/setEnded", methods={"POST"})
+    * @AccessControl(role="INSTRUCTOR")
+    * @return MultiResponse
+    */
+    public function endPoll() {
+        $this->core->getQueries()->endPoll($_POST["poll_id"]);
+
+        return MultiResponse::RedirectOnlyResponse(
+            new RedirectResponse($this->core->buildCourseUrl(['polls']))
+        );
+    }
+
+    /**
+    * @Route("/{_semester}/{_course}/polls/submitResponse", methods={"POST"})
+    * @return MultiResponse
+    */
+    public function submitResponse() {
+        $poll = $this->core->getQueries()->getPoll($_POST["poll_id"]);
+        if ($poll == null) {
+            $this->core->addErrorMessage("Invalid Poll ID");
+            return MultiResponse::RedirectOnlyResponse(
+                new RedirectResponse($this->core->buildCourseUrl(['polls']))
+            );
+        }
+        if ($poll->isOpen()) {
+            $this->core->getQueries()->submitResponse($_POST["poll_id"], $_POST["answer"]);
+        }
+
+        return MultiResponse::RedirectOnlyResponse(
+            new RedirectResponse($this->core->buildCourseUrl(['polls']))
+        );
+    }
+
+    /**
+     * @Route("/{_semester}/{_course}/polls/editPoll", methods={"POST"})
+     * @AccessControl(role="INSTRUCTOR")
+     * @return MultiResponse
+     */
+    public function editPoll() {
+        $poll = $this->core->getQueries()->getPoll($_POST["poll_id"]);
+
+        if ($poll == null) {
+            $this->core->addErrorMessage("Invalid Poll ID");
+            return MultiResponse::RedirectOnlyResponse(
+                new RedirectResponse($this->core->buildCourseUrl(['polls']))
+            );
+        }
+
+        return MultiResponse::webOnlyResponse(
+            new WebResponse(
+                'Poll',
+                'editPoll',
+                $poll
+            )
+        );
+    }
+
+    /**
+     * @Route("/{_semester}/{_course}/polls/editPoll/submitEdits", methods={"POST"})
+     * @AccessControl(role="INSTRUCTOR")
+     * @return MultiResponse
+     */
+    public function submitEdits() {
+        $response_count = $_POST["response_count"];
+        $responses = array();
+        $answers = array();
+        $orders = array();
+        for ($i = 0; $i < $response_count; $i++) {
+            $responses[$_POST["option_id_" . $i]] = $_POST["response_" . $i];
+            $orders[$_POST["option_id_" . $i]] = $_POST["order_" . $i];
+            if (isset($_POST["is_correct_" . $i]) && $_POST["is_correct_" . $i] == "on") {
+                $answers[] = $_POST["option_id_" . $i];
+            }
+        }
+        $this->core->getQueries()->editPoll($_POST["poll_id"], $_POST["name"], $_POST["question"], $responses, $answers, $_POST["release_date"], $orders);
```

Comment:
You should add some error checking in case there is no `name`, `question` and `release_date`